source/MaterialXTest/ShaderValid.cpp

```diff
@@ -0,0 +1,291 @@
+        }
+        catch (mx::ExceptionShaderValidationError e)
+        {
+            for (auto error : e._errorLog)
+            {
+                std::cout << e.what() << " " << error << std::endl;
+            }
+        }
+        REQUIRE(uniformsParsed);
+
+        bool attributesParsed = false;
+        try
+        {
+            const mx::GlslValidator::ProgramInputMap& attributes = validator.getAttributesList();
+            for (auto input : attributes)
+            {
+                unsigned int type = input.second->_type;
+                int location = input.second->_location;
+                int size = input.second->_size;
+                std::cout << "Program Attribute: \"" << input.first << "\". Location=" << location << ". Type=" << std::hex << type << ". Size=" << size << "." << std::endl;
+            }
+            attributesParsed = true;
+        }
+        catch (mx::ExceptionShaderValidationError e)
+        {
+            for (auto error : e._errorLog)
+            {
+                std::cout << e.what() << " " << error << std::endl;
+            }
+        }
+        REQUIRE(attributesParsed);
+
+        bool renderSucceeded = false;
+        try
+        {
+            validator.render();
+            renderSucceeded = true;
+        }
+        catch (mx::ExceptionShaderValidationError e)
+        {
+            for (auto error : e._errorLog)
+            {
+                std::cout << e.what() << " " << error << std::endl;
+            }
+        }
+        REQUIRE(renderSucceeded);
+
+        try
+        {
+            mx::TinyEXRImageHandlerPtr handler = std::make_shared<mx::TinyEXRImageHandler>();
+            std::string fileName = shaderName + ".exr";
+            validator.save(fileName, handler);
+        }
+        catch (mx::ExceptionShaderValidationError e)
+        {
+            for (auto error : e._errorLog)
+            {
+                std::cout << e.what() << " " << error << std::endl;
+            }
+        }
+    }
+}
+
+
+TEST_CASE("GLSL Validation from HwShader", "[view]")
+{
+    mx::DocumentPtr doc = mx::createDocument();
+
+    // Load standard libraries
+    std::vector<std::string> filenames =
+    {
+        "documents/Libraries/stdlib/mx_stdlib_defs.mtlx",
+        "documents/Libraries/stdlib/impl/shadergen/glsl/impl.mtlx"
+    };
+
+    for (const std::string& filename : filenames)
+    {
+        mx::readFromXmlFile(doc, filename);
+    }
+
+    mx::FilePath searchPath = mx::FilePath::getCurrentPath() / mx::FilePath("documents/Libraries");
+    mx::ShaderGeneratorPtr shaderGenerator = mx::GlslShaderGenerator::creator();
+    shaderGenerator->registerSourceCodeSearchPath(searchPath);
+
+    // Create a nonsensical graph testing some geometric nodes
+    mx::NodeGraphPtr nodeGraph = doc->addNodeGraph("geometry_attributes");
+
+    std::vector<mx::NodePtr> attributeList;
+
+    // normal output test
+    mx::NodePtr normal1 = nodeGraph->addNode("normal", "normal1", "vector3");
```

Comment:
This dumps out images for attributes / textures which can be looked at to check if more complex shader graphs do the right thing. There is no image compare currently.